File: server/SE.Catalog.API/SE.Catalog.Models/Package.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace SE.Catalog.Models
{
    public class Package : BaseEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string ProductName { get; set; }
        public string HWVersion { get; set; }

        public int DeviceFamilyId { get; set; }
        public int ProductFamilyId { get; set; }

        [ForeignKey("ProductFamilyId")]
        public ProductFamily ProductFamily { get; set; }
        [ForeignKey("DeviceFamilyId")]
        public DeviceFamily DeviceFamily { get; set; }

        public int VendorId { get; set; }
        [ForeignKey("VendorId")]
        public Vendor Vendor { get; set; }
        public string ProfileType { get; set; }
        public WorkFlowStatus Status { get; set; }
        public int OwnerId { get; set; }
        public DateTime UploadDate { get; set; }
        public string ProductId { get; set; }
        public string SWVersion { get; set; }
        public string FileName { get; set; }
        public string BlobURL { get; set; }
        public List<PackageComment> Comments { get; set; }
        
        
        
    }
    public enum WorkFlowStatus
    {
        Approved,
        Pending,
        Rejected
    }
}

File: client/CatalogManager/src/app/models/package.ts
import { IDevice } from "./device";

export interface IPackage{

    id: Int32Array;
    checked: boolean;
    name: string;
    deviceFamilyId: string;
    productFamilyId: string;
    status: number;
    productFamily: any;
    deviceFamily: any;
    productName: string;
    OrderNumber: string;
    hwVersion: string;
    profileType: string;
    ownerId: string;
    uploadDate: Date;
    productId: string;
    swVersion: string;
    fileName: string;
    blobURL: string;
    comments: string;
    lastModified: Date;
    createdOn: Date;
}
